x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/action/GetLifecycleAction.java

```diff
@@ -96,6 +83,23 @@ public String toString() {
             return Strings.toString(this, true, true);
         }
 
+        @Override
+        @SuppressWarnings("unchecked")
+        public Iterator<? extends ToXContent> toXContentChunked(ToXContent.Params outerParams) {
+            return Iterators.concat(
+                Iterators.single((ToXContent) (builder, params) -> builder.startObject()),
+                Iterators.concat(policies.stream().map(policy -> Iterators.single((ToXContent) (b, p) -> {
+                    b.startObject(policy.getLifecyclePolicy().getName());
+                    b.field("version", policy.getVersion());
+                    b.field("modified_date", policy.getModifiedDate());
+                    b.field("policy", policy.getLifecyclePolicy());
+                    b.field("in_use_by", policy.getUsage());
+                    b.endObject();
+                    return b;
+                })).toArray(Iterator[]::new)),
```

Comment:
Maybe this is a little nicer?

```suggestion
        public Iterator<? extends ToXContent> toXContentChunked(ToXContent.Params outerParams) {
            return Iterators.concat(
                Iterators.single((builder, params) -> builder.startObject()),
                policies.stream().map(policy -> (ToXContent) (b, p) -> {
                    b.startObject(policy.getLifecyclePolicy().getName());
                    b.field("version", policy.getVersion());
                    b.field("modified_date", policy.getModifiedDate());
                    b.field("policy", policy.getLifecyclePolicy());
                    b.field("in_use_by", policy.getUsage());
                    b.endObject();
                    return b;
                }).iterator(),
```

Reply:
Ah sweet, TIL